### PR TITLE
Add a blogpost on exploring the IntelliJ plugin's sync process

### PR DESCRIPTION
**Pull request by jin**

### CODE REVIEW

**Comment by jin** [comment]
cc @chaoren @brendandouglas

File: _posts/2019-09-19-ijwb-reverse-engineering-sync.md [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+      is_external: false
+      is_new_external_version: true
+      is_source: false
+      relative_path: "src/main/java/hello/app.jar"
+      root_execution_path_fragment: "bazel-out/darwin-fastbuild/bin"
+    }
+    source_jar {
+      is_external: false
+      is_new_external_version: true
+      is_source: false
```
Comment (ittaiz) [inline _posts/2019-09-19-ijwb-reverse-engineering-sync.md]: how is this false? Also below? When is this true? Maybe it's not important to this post, your call

Comment (jin) [reply _posts/2019-09-19-ijwb-reverse-engineering-sync.md]: Thanks. Adding something for 1).

2-5 is out of scope for my intention of this post. The working set is already [documented](https://ij.bazel.build/docs/bazel-plugin.html#expand-sync-to-working-set), and `derive_targets_from_directories` is a relatively new feature for project management that is not directly impacting how the sync process works internally.

Comment (jin) [reply _posts/2019-09-19-ijwb-reverse-engineering-sync.md]: `is_source` is true only for artifacts that are source code to your project. e.g. genrule outputs are not `is_source`. It has nothing to do with source jars.

File: _posts/2019-09-19-intellij-bazel-sync.md [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+  # python
+  # scala
+```
+
+## Bazel data subdirectory
+
+`.ijwb/.blaze` is the Bazel data subdirectory, containing mostly IntelliJ module
+definitions.
+
+Note that the actual persistent serialized data is not in this directory. To
```
Comment (brendandouglas) [inline _posts/2019-09-19-intellij-bazel-sync.md]: Much of the persisted serialized data in this directory -- might be worth rephrasing this.

Comment (jin) [reply _posts/2019-09-19-intellij-bazel-sync.md]: done. thanks!

**Comment by jin** [comment]
@dslomov @aiuto @laurentlb could one of you ptal at this? received lgtm from @brendandouglas for technical accuracy.

**Pull request merged**